gossip/c_block_callbacks.go

```diff
@@ -262,6 +268,14 @@ func consensusCallbackBeginBlockFn(
 							}
 						}
 					}
+					// Skipped txs require original order from events
+					skippedTxs := make([]uint32, 0, len(orderedTxs))
+					for _, txNumber := range skippedByEvm {
+						txHash := orderedTxs[txNumber].Hash()
+						skippedTxs = append(skippedTxs, txPositions[txHash].EventOffset)
+					}
+					// Txs skipped by scrambler holds on the original order from events
+					block.SkippedTxs = append(skippedTxs, skippedByScrambler...)
```

Comment:
Any idea how to test this? I think this code is too critical and complex to go without testing ...

Reply:
I guess we can create a helper func `getSkippedTransactions` and test it through that.